cfme/tests/intelligence/reports/test_import_export_reports_widgets.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme import test_requirements
+from cfme.intelligence.reports.reports import ImportExportCustomReportsView
+from cfme.intelligence.reports.widgets import ImportExportWidgetsCommitView
+from cfme.utils.path import data_path
+
+pytestmark = [pytest.mark.tier(1), test_requirements.report]
+
+
+@pytest.fixture()
+def widget_file(appliance):
+    file_path = data_path.join("ui/intelligence/import_widget.yaml").realpath().strpath
+    widget = appliance.collections.dashboard_report_widgets.instantiate(
+        appliance.collections.dashboard_report_widgets.CHART,
+        "testing widget",
+        description="testing widget description",
+        filter="Configuration Management/Virtual Machines/Guest OS Information - any OS",
+        active=True,
+    )
+
+    yield file_path, widget
+
+    # delete the widget in case it was imported or created
+    widget.delete_if_exists()
+
+
+@pytest.fixture(scope="function")
+def report_file(appliance):
+    file_path = data_path.join("ui/intelligence/import_report.yaml").realpath().strpath
+    report = appliance.collections.reports.instantiate(
+        type="My Company (All Groups)",
+        subtype="Custom",
+        menu_name="testing report",
+        title="testing report title",
+    )
+
+    yield file_path, report
+
+    # delete the report in case it was imported or created
+    report.delete_if_exists()
+
+
+def test_import_widget(appliance, widget_file):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        initialEstimate: 1/16h
+        startsin: 5.3
+        testSteps:
+            1. Import the widget data yaml.
+            2. Check if widget created with import is same as the expected widget.
+    """
+    file_path, widget = widget_file
+    collection = appliance.collections.dashboard_report_widgets
+
+    collection.import_widget(file_path)
+    import_view = collection.create_view(ImportExportWidgetsCommitView)
+    import_view.flash.assert_message("1 widget imported successfully")
+
+    assert widget.exists
+
+
+def test_export_widget(appliance, widget_file):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        initialEstimate: 1/16h
+        startsin: 5.3
+    """
+    _, widget = widget_file
+    collection = appliance.collections.dashboard_report_widgets
+    collection.create(
+        widget_class=getattr(collection, widget.TITLE.upper()),
+        title=widget.title,
+        description=widget.description,
+        filter=widget.filter,
+        active=widget.active,
+    )
+    collection.export_widget(widget.title)
```

Comment:
This needs additional assertion/verification. The `export_widget` method itself only asserts there is no error type flash message, is there a success flash message to be validated here? Can we further verify the result of selecting widget export?

Reply:
There is no message displayed when we export the widget, it just gets exported. I don't think there is a sure-shot way of further testing this. If we could check the files downloaded, we could further verify it, but doing that seems more trouble than worth for.